src/firefly/js/ui/FileUpload.jsx

```diff
@@ -63,27 +63,27 @@ function getProps(params, fireValueChange) {
     return Object.assign({}, params,
         {
             value: params.displayValue,
-            onChange: (ev) => handleChange(ev, fireValueChange, params.fileType)
+            onChange: (ev) => handleChange(ev, fireValueChange, params.fileType, params.fileAnalysis)
         });
 }
 
-function handleChange(ev, fireValueChange, type) {
+function handleChange(ev, fireValueChange, type, fileAnalysis) {
     var file = get(ev, 'target.files.0');
     var displayValue = get(ev, 'target.value');
 
     fireValueChange({
         displayValue,
-        value: makeDoUpload(file, type)
+        value: !fileAnalysis ? makeDoUpload(file, type) : makeDoUpload(file, type, fileAnalysis)()
     });
 }
 
-function makeDoUpload(file, type) {
+function makeDoUpload(file, type, fileAnalysis) {
```

Comment:
When the file is uploading we need to mask the whole dialog.  Look at line 98 of ImageSelectDropdown.jsx`. 

Use `className='loading-mask'`